### PR TITLE
BUG: is_string_dtype returns True for ArrowDtype(pa.string())

### PR DESCRIPTION
**Pull request by mroeschke**
Broken off #50325

### CODE REVIEW

File: pandas/tests/extension/test_arrow.py [hunk 1]
```diff
@@ -743,6 +743,13 @@ def test_get_common_dtype(self, dtype, request):
             )
         super().test_get_common_dtype(dtype)
 
+    def test_is_not_string_type(self, dtype):
+        pa_dtype = dtype.pyarrow_dtype
+        if pa.types.is_string(pa_dtype):
```
Comment (jbrockmendel) [inline pandas/tests/extension/test_arrow.py:748]: should the binary types be considered stringy?

Comment (mroeschke) [reply pandas/tests/extension/test_arrow.py:748]: I would say no as of now, as `pa.types.is_string` doesn't consider binary striny

Comment (jbrockmendel) [reply pandas/tests/extension/test_arrow.py:766]: fair enough

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @mroeschke

**Pull request merged**